src/atlas_f/grid/atlas_Grid_module.F90

```diff
@@ -553,6 +561,65 @@ function Structured__ny(this) result(ny)
   ny = atlas__grid__Structured__ny(this%CPTR_PGIBUG_A)
 end function
 
+subroutine Structured__index2ij_int32(this, gidx, i, j) 
+  use, intrinsic :: iso_c_binding, only: c_long, c_int, c_loc
+  use atlas_grid_Structured_c_binding
+  integer(c_int), intent (in) :: gidx
+  class(atlas_StructuredGrid), intent(in) :: this
+  integer(c_int), intent(out) :: i, j
+  call atlas__grid__Structured__index2ij(this%CPTR_PGIBUG_A, int (gidx-1, c_long), c_loc (i), c_loc(j))
```

Comment:
I noticed some compile errors with this. Normally it should work **without** c_loc
if you make the C interface function with `int& i, int& j` in the signature instead of `int* i, int* j`